crates/kornia-io/src/jpeg.rs

```diff
@@ -0,0 +1,201 @@
+    image: &Image<u8, N>,
+    color_type: ExtendedColorType,
+) -> Result<(), IoError> {
+    let image_size = image.size();
+    let file = File::create(file_path)?;
+    let encoder = JpegEncoder::new(file);
+    encoder.write_image(
+        image.as_slice(),
+        image_size.width as u32,
+        image_size.height as u32,
+        color_type,
+    )?;
+    Ok(())
+}
+
+/// Read a JPEG image with a four channel _(rgb8)_.
+///
+/// # Arguments
+///
+/// - `file_path` - The path to the JPEG file.
+///
+/// # Returns
+///
+/// A RGB image with four channels _(rgb8)_.
+pub fn read_image_jpeg_rgb8(file_path: impl AsRef<Path>) -> Result<Image<u8, 3>, IoError> {
+    read_image_jpeg_internal(file_path, PixelFormat::RGB24)
+}
+
+/// Reads a JPEG file with a single channel _(mono8)_
+///
+/// # Arguments
+///
+/// - `file_path` - The path to the JPEG file.
+///
+/// # Returns
+///
+/// A grayscale image with a single channel _(mono8)_.
+pub fn read_image_mono8(file_path: impl AsRef<Path>) -> Result<Image<u8, 1>, IoError> {
+    read_image_jpeg_internal(file_path, PixelFormat::L8)
+}
+
+/// Reads a JPEG file with a single channel _(mono16)_.
+///
+/// # Arguments
+///
+/// - `file_path` - The path to the JPEG file.
+///
+/// # Returns
+///
+/// A grayscale image with a single channel _(mono16)_.
+pub fn read_image_mono16(file_path: impl AsRef<Path>) -> Result<Image<u16, 1>, IoError> {
+    let image: Image<u8, 1> = read_image_jpeg_internal(file_path, PixelFormat::L16)?;
+    let image_size = image.size();
+    let image_buf = image.as_slice();
+
+    let mut buf = Vec::with_capacity(image_buf.len() / 2);
+    for img in image_buf {
+        buf.push(*img as u16);
+    }
+
+    Ok(Image::new(image_size, buf)?)
+}
+
+fn read_image_jpeg_internal<const N: usize>(
+    file_path: impl AsRef<Path>,
+    color_type: PixelFormat,
+) -> Result<Image<u8, N>, IoError> {
+    let file_path = file_path.as_ref().to_owned();
+    if !file_path.exists() {
+        return Err(IoError::FileDoesNotExist(file_path.to_path_buf()));
+    }
+
+    if file_path.extension().map_or(true, |ext| {
+        !ext.eq_ignore_ascii_case("jpg") && !ext.eq_ignore_ascii_case("jpeg")
+    }) {
+        return Err(IoError::InvalidFileExtension(file_path.to_path_buf()));
+    }
+
+    let jpeg_data = File::open(file_path)?;
+    let mut decoder = jpeg_decoder::Decoder::new(jpeg_data);
+    decoder.read_info().map_err(|e| IoError::JpegError(e))?;
+
+    let image_info = decoder.info().ok_or_else(|| {
+        IoError::JpegError(jpeg_decoder::Error::Format(String::from(
+            "Failed to found Image Info from it's metadata",
+        )))
+    })?;
+
+    let pixel_format = image_info.pixel_format;
+
+    let image_size = ImageSize {
+        width: image_info.width as usize,
+        height: image_info.height as usize,
+    };
+
+    let img_data = decoder.decode()?;
+
+    // Convert the image color type, if necessary
+    if color_type != pixel_format {
```

Comment:
I'm not sure about this. I would make the assumption for now that the user knows or has enough documentation to properly the function by passing the right image. At least we raise an error. In the worst case I believe the user can perform themselves a re-enconding with proper error handling. Let's keep the function simple